packages/plugin-kit/src/lib/decodeSliceMachineConfig.ts

```diff
@@ -0,0 +1,39 @@
+import * as t from "io-ts";
+
+import { SliceMachineConfig } from "../types";
+
+import { decode, DecodeReturnType } from "./decode";
+
+const SliceMachineConfigPluginRegistrationCodec = t.union([
+	t.string,
+	t.intersection([
+		t.type({
+			resolve: t.string,
+		}),
+		t.partial({
+			options: t.UnknownRecord,
+		}),
+	]),
+]);
+
+const SliceMachineConfigCodec = t.intersection([
+	t.type({
+		_latest: t.string,
+		repositoryName: t.string,
+		adapter: SliceMachineConfigPluginRegistrationCodec,
+	}),
+	t.partial({
+		apiEndpoint: t.string,
+		libraries: t.array(t.string),
+		localSliceSimulatorURL: t.string,
+		plugins: t.array(SliceMachineConfigPluginRegistrationCodec),
+	}),
+]);
+
+// TODO: Maybe rename "decode" to "validate". "decode" exposes the `io-ts`
```

Comment:
It would make it more user friendly

Reply:
Since it's not a public API, I think it's fine to preserve `io-ts` semantic, the manager uses `decode*` too